core/src/main/scala/org/broadinstitute/dsde/rawls/workspace/WorkspaceService.scala

```diff
@@ -701,20 +708,36 @@ class WorkspaceService(protected val ctx: RawlsRequestContext,
         fastPassServiceConstructor(childContext, dataSource).removeFastPassGrantsForWorkspace(workspaceContext)
       )
 
-      // Delete Google Project
+      // Delete Google Project but not the project's Sam record
       _ <- traceFutureWithParent("maybeDeleteGoogleProject", parentContext)(_ =>
-        if (!isAzureMcWorkspace(maybeMcWorkspace)) {
-          maybeDeleteGoogleProject(workspaceContext.googleProjectId,
-                                   workspaceContext.workspaceVersion,
-                                   ctx.userInfo
-          ) recoverWith { case t: Throwable =>
-            logger.error(
-              s"Unexpected failure deleting workspace (while deleting google project) for workspace `${workspaceContext.toWorkspaceName}`",
-              t
-            )
-            Future.failed(t)
-          }
-        } else Future.successful()
+        maybeDeleteGoogleProject(workspaceContext.googleProjectId,
```

Comment:
This is the crux of the change. We are deleting the google cloud project (but not the sam resource as we still need the permissions so leo's sam assertions still work). Once leo has cleaned up it's resource records, we can clear out the google project sam resource.